### DISCUSSION TITLE
Design choice regarding bootstrapping in MAGMA

### DISCUSSION THREAD

**Discussion by abhi18av** [TORCH-Consortium/MAGMA#164] [General]
Dear @vrennie @TimHHH @mdediegofuertes 

I came across a question regarding bootstrapping in the pipeline, which we don't use as of now. Previously, we have added some comments but perhaps we should outline when and why exactly one might need to tweak the `IQTREE` arguments

https://github.com/TORCH-Consortium/MAGMA/blob/e5011cbb8675e82ef626687733b00491f5da6ef8/default_params.config#L631-L633 

The final arguments with which we run `IQTREE` at this point are as follows 

```
iqtree \
    -s Filtering_Test.variable.ExDR.IncComplex.fa \
    -T AUTO \
     -allnni -redo  \
    --prefix Filtering_Test.ExDR.IncComplex

```

I am initiating this discussion on Github itself document the design choice better.

**Comment by TimHHH** [comment]
1.  ```-allnni``` ensures a very thorough search of possible phylogenies, in certain cases too thorough, if you find that IQTREE consistently finds the same phylogeny for your dataset with and without this option then you may exclude it to speed up phylogenetic inference.
2. bootstrapping is used to infer a certain level of confidence (out of 100) for each split in the phylogeny, this is done by resampling and reanalysing the fasta, this standard bootstrap can be called with the ```-b 500``` option and the bootstrap scores are included in the phylogentic outputs. >=500 recommended, WARNING: this will slow down time to results massively.
3. fast bootstrapping uses the method described [here](https://academic.oup.com/mbe/article/30/5/1188/997508). Fast bootstrapping is done with ```-bb 1000```, >=1000 recommended, WARNING: this will slow down time to results. Consider including ```-alrt 1000``` to perform [the recommended SH-aLRT test](http://www.iqtree.org/doc/Frequently-Asked-Questions#how-do-i-interpret-ultrafast-bootstrap-ufboot-support-values)

This is not something one should do if not familiar with the interpretation of bootstraps, this is not something for clinicians.

**Reply by abhi18av** [reply]
Thanks @TimHHH  for the answer.

> This is not something one should do if not familiar with the interpretation of bootstraps, this is not something for clinicians.

My question is specially around this phrase, perhaps the research groups who are especially focused on phylogenetic trees - they might benefit from `-alrt 1000` right?

If time consumption by the `IQTREE` process is a concern, then perhaps we can introduce a couple of flags like `--fast-phylogeny` and `--standard-boostrap` ?

Though, users can just customise using a config file so the params thing is just an idea.

**Comment by TimHHH** [comment]
> If time consumption by the `IQTREE` process is a concern, then perhaps we can introduce a couple of flags like `--fast-phylogeny` and `--standard-boostrap` ?

If we can give each flag a section in the default parameters file then I think this would be great. Such a section would then include the line with all commands for the given flag so that users can change the parameters. Something along these lines:

```
#accurate maximumlikelihood phylogeny only DEFAULT
--accurate-ml-only = '-allnni'

#fast maximumlikelihood phylogeny only
--fast-ml-only = '-fast'

#fast-bootstrapped-phylogeny
--fast-bs-phylogeny = '-bb1000 -alrt 1000'

#standard-boostrap
--standard-boostrap = '-bb1000'
```

maybe also reflect the used flag as part of the output prefix.

**Reply by vrennie** [reply]
Sounds good to me!

**Reply by abhi18av** [reply]
Oki doki, I've now added 4 `iqtree_*` specific parameters which reflect the scheme outlined earlier 

https://github.com/TORCH-Consortium/MAGMA/pull/170/commits/1e9c7027f2e9319a879a16a844eb01df3a27707b

**Comment by vrennie** [comment]
Hi @abhi18av can you inform me where I need do add the --standard-boostrap = '-bb1000' if I am launching on Nextflow? I tried it here but I don't think it worked:

https://tower.nf/orgs/TORCH/workspaces/XBS-Nextflow/watch/v6E4UB2ccO5Qt

**Reply by abhi18av** [reply]
Hey @vrennie the new parameters are added as 

1. On CLI `--iqtree_standard_bootstrap`, you need to use the flag
2. In the params file, you need to use `iqtree_standard_bootstrap: true` 

Please check here for other options https://github.com/TORCH-Consortium/MAGMA/discussions/164#discussioncomment-6904235

**Comment by TimHHH** [comment]
The bootstraps can be found inside the *.treefile . 

Figtree will say that it has identified some undefined label, you can give it the name ‘bootstrap’ or whatever you like.

The bootstrap labels can then be visualised by enabling node labels and specifying that the ‘bootstrap’ has to be displayed there.
